File: android/app/src/main/kotlin/com/bockmedia/console/domain/model/LibrarySessionCache.kt
package com.bockmedia.console.domain.model

/** In-memory library buckets — filter switches are instant; refreshed in background. */
object LibrarySessionCache {
    private const val TTL_MS = 600_000L

    @Volatile
    var data: LibraryData? = null
        private set

    @Volatile
    private var cachedAtMs: Long = 0L

    fun getIfFresh(): LibraryData? {
        val cached = data ?: return null
        if (System.currentTimeMillis() - cachedAtMs > TTL_MS) return null
        return cached
    }

    fun peek(): LibraryData? = data

    fun put(value: LibraryData) {
        if (value.playlists.isEmpty() && value.artists.isEmpty() && value.albums.isEmpty()) return
        data = value
        cachedAtMs = System.currentTimeMillis()
    }

    fun invalidate() {
        data = null
        cachedAtMs = 0L
    }
}

File: ios/BockMedia/Domain/LibrarySessionCache.swift
import Foundation

enum LibrarySessionCache {
    private static let ttlMs: Int64 = 90_000
    private static var data: LibraryData?
    private static var cachedAtMs: Int64 = 0

    static func getIfFresh() -> LibraryData? {
        guard let cached = data else { return nil }
        if Int64(Date().timeIntervalSince1970 * 1000) - cachedAtMs > ttlMs { return nil }
        return cached
    }

    static func peek() -> LibraryData? { data }

    static func put(_ value: LibraryData) {
        if value.playlists.isEmpty && value.artists.isEmpty && value.albums.isEmpty { return }
        data = value
        cachedAtMs = Int64(Date().timeIntervalSince1970 * 1000)
    }

    static func invalidate() {
        data = nil
        cachedAtMs = 0
    }
}
